apps/xprof_gui/priv/src/actions/CollectingActions.js

```diff
@@ -29,11 +29,13 @@ export const getMonitoredFunctions = () => async (dispatch, getState) => {
   if (error) {
     console.log('ERROR: ', error);
   } else if (!isEqual(mfas, json)) {
-    const newMfas = json.filter(mfa => !mfas.map(f => f[3]).includes(mfa[3]));
+    const newMfas = json
+      .filter(mfa => !mfas.map(f => f.query)
```

Comment:
nitpicking: you could perform and collect `mfas.map(f => f.query)` before filtering to do it only once instead of doing it for each `mfa` from incomming `json`

Reply:
Now it is:

```
const queries = mfas.map(f => f.query);
const newMonitoredCollection = json
  .filter(mfa => !queries
    .includes(mfa.query));
```

(for the sake of clarity, given in old, already changed naming convention)